doc/source/whatsnew/v1.0.0.rst

```diff
@@ -314,6 +314,7 @@ I/O
 - Bug in :func:`read_hdf` closing stores that it didn't open when Exceptions are raised (:issue:`28699`)
 - Bug in :meth:`DataFrame.read_json` where using ``orient="index"`` would not maintain the order (:issue:`28557`)
 - Bug in :meth:`DataFrame.to_html` where the length of the ``formatters`` argument was not verified (:issue:`28469`)
+- Bug in :meth:`DataFrame.to_json` where timezone-aware dates were converted to UTC (:issue:`12997`)
```

Comment:
should this specify that it is only fixed-offset timezones that are now supported?  any other names e.g. pd.to_json to mention?

Reply:
I actually don't know what fixed versus (assumedly) non-fixed offsets are - can you clarify?

Reply:
"UTC-8" is a fixed offset.  "US/Pacific" (or colloquially-but-inaccurately "PST/PDT") is not a fixed offset.

Reply:
So I don't think that distinction matters when writing then (?). At least in tests the construction goes something like `pd.date_range(..., tz="US/Eastern")` and it writes out in ISO format

That may be a problem when reading. There needs to be a follow up on the reading side to support this as this now does the following when using "+0000" instead of "Z" as the time zone designator

```python
In [5]: df = pd.DataFrame(pd.date_range("2013-01-01", periods=2, tz="US/Eastern"), columns=['date'])

In [6]: pd.read_json(df.to_json(date_format="iso")).dtypes
Out[6]:
date    datetime64[ns, pytz.FixedOffset(-300)]
dtype: object
```

Reply:
Ah never mind...I get what you are saying now. I guess the ISO format is ambiguous as to what timezone it is, so sure fixed only. Can update that on next iteration

Reply:
i guess if youve specified date_format="iso" then the user knows what they're getting into.  its just annoying that we dont have a nice way to round-trip read/write non-fixed-offset timezones